src/test/java/com/echoserver/test/EchoServerTest.java

```diff
@@ -1,13 +1,36 @@
-
 package com.echoserver.test;
 
 import com.echoserver.EchoServer;
+import org.junit.Assert;
 import org.junit.Test;
 
+import java.io.*;
+
 public class EchoServerTest {
+    private EchoServer server;
+    private StringWriter out;
+
     @Test
-    public void test() {
-        EchoServer server = new EchoServer("my server");
-        assert server.getName().equals("my server");
+    public void exitTheEchoServerWithoutOutput() throws IOException {
+        initialiseEchoServerWithInput("exit\n");
+        server.run();
+
+        Assert.assertEquals(out.toString(), "");
```

Comment:
It's also possible the `static import` these methods, so that the `Assert.` prefix can be removed.